src/screens/FavoritesScreen.js

```diff
@@ -14,22 +22,106 @@ class FavoritesScreen extends Component {
             title: 'Favorites',
             headerTitle: 'Your Favorite Campsites',
             headerLeft: null,
-            tabBarIcon: ({focused, tintColor }) => (<Icon type='ionicon' name={focused ? 'ios-heart' : 'ios-heart-outline'} size={25} color={tintColor} />)
+            tabBarIcon: ({focused, tintColor}) => (
+                <Icon type='ionicon' name={focused ? 'ios-heart' : 'ios-heart-outline'} size={25} color={tintColor}/>)
         }
 
     };
 
-    render(){
-        return(
-            <Card>
-                <CardSection>
-                    <Text>
-                        Favorite sites go here!
-                    </Text>
-                </CardSection>
-            </Card>
+    onPressFacebookLogin = () => {
+        const {navigation: {navigate}} = this.props;
+
+        this.props.logUserIntoFacebook({navigate});
+    };
+
+    renderFavorites = ({sites, getSiteDetail, navigate}) => {
+        return _.map(sites, (site, index) => {
+            return (
+                <CampsiteListItem
+                    key={index}
+                    site={site}
+                    getSiteDetail={getSiteDetail}
+                    navigate={navigate}
+                />
+            );
+        });
+    };
+
+    renderFavoritesScreen = () => {
+        const {currentUser} = this.props;
+        const {headerTitleStyle, infoTextStyle, facebookStyle} = styles;
+
+        if (!currentUser || currentUser.name === tokens.GUEST) {
+            return (
+                <Card>
+                    <Text style={headerTitleStyle}>{common.must_log_in}</Text>
+                    <Text style={infoTextStyle}>{must_log_in_detail}</Text>
+
+                    <Button
+                        large
+                        transparent
+                        icon={{name: 'facebook', type: 'font-awesome'}}
+                        title={login.login_with_facebook}
+                        buttonStyle={facebookStyle}
+                        onPress={this.onPressFacebookLogin}
+                        rounded={true}
+                    />
+                </Card>
```

Comment:
If the user is logged in as Guest, let them know that they need to fully log in to add favorites. The button here does the same logic that we have on the `LoginScreen` to log the user in.